File: BudgetApp.API/Models/BudgetLineItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace BudgetApp.API.Models
{
    public partial class BudgetLineItem
    {
        public int Id { get; set; }
        public string AFE_Id { get; set; } = null!;
        public int Gate { get; set; }
        public int WBS_Id { get; set; }
        public int? Area_Id { get; set; }
        public int? Unit_Id { get; set; }
        public string? Vendor { get; set; }
        public double? Quantity { get; set; }
        public decimal Unit_Cost { get; set; }
        public double? Tax_Rate { get; set; }
        public decimal Total_Cost { get; set; }
        public string? Comment { get; set; }
        public DateTime Date_Created { get; set; }
        public string Created_By { get; set; } = null!;
        public DateTime? Date_Modified { get; set; }
        public string? Modified_By { get; set; }
        public int Parent_Id { get; set; }
        public int Revision_No { get; set; }
        public bool Is_Active { get; set; }

        [ForeignKey("AFE_Id")]
        public virtual AFE AFE { get; set; } = null!;
        [ForeignKey("Area_Id")]
        public virtual Area? Area { get; set; } = null!;
        [ForeignKey("Unit_Id")]
        public virtual Unit? Unit { get; set; } = null!;
        [ForeignKey("WBS_Id")]
        public virtual WBS_Dictionary WBS { get; set; } = null!;
    }
}

File: BudgetApp-SPA/src/app/models/budget-line-item.ts
export interface BudgetLineItem {
  id?: number;
  afE_Id: string;
  gate: number;
  wbS_Id: number;
  area_Id: number;
  unit_Id: number;
  vendor?: string;
  quantity: number;
  unit_Cost: number;
  tax_Rate: number;
  total_Cost: number;
  comment?: string;
  date_created?: Date;
  created_by?: string;
  date_modified?: Date;
  modified_by?: string;
  revision_no?: number;
  is_active?: boolean;
  area?: any;
  unit?: any;
  wbs?: any;
  wbsForm?: any;
  parent_Id?: number;
}
